{
  "modified": "2020-01-21T21:06:18Z",
  "published": "2016-09-02T14:00:00Z",
  "schema_version": "1.5.0",
  "id": "CVE-2016-5636",
  "aliases": [
    "PSF-2016-7"
  ],
  "details": "Integer overflow in the get_data function in zipimport.c in CPython (aka Python) before 2.7.12, 3.x before 3.4.5, and 3.5.x before 3.5.2 allows remote attackers to have unspecified impact via a negative data size value, which triggers a heap-based buffer overflow.",
  "affected": [
    {
      "package": {
        "ecosystem": "DHI",
        "name": "python",
        "purl": "pkg:dhi/python"
      },
      "ranges": [
        {
          "type": "SEMVER",
          "events": [
            {
              "introduced": "2.7.0"
            },
            {
              "fixed": "2.7.12"
            },
            {
              "fixed": "3.10.0-alpha1"
            },
            {
              "introduced": "3.10.0"
            }
          ],
          "repo": "https://github.com/python/cpython"
        }
      ]
    }
  ],
  "references": [
    {
      "type": "REPORT",
      "url": "https://bugs.python.org/issue26171"
    }
  ]
}